coralproject/talk
# [CORL-2861] add left mod queue web hook option

## What does this PR do?

Adds COMMENT_LEFT_MODERATION_QUEUE option to `Admin > Configure > Webhooks`.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

- Adds `COMMENT_LEFT_MODERATION_QUEUE` to `WEBHOOK_EVENT_NAME` enum.

## Does this PR introduce any new environment variables or feature flags?

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indices.

## How do I test this PR?

- spin up a quick express  + cors middleware web server to act as a web hook listener, the following script works nicely:
```
const express = require("express");
const cors = require("cors");

const PORT = 7000;
const HOST = "localhost";

const run = async () => {
  const app = express();

  app.use(cors());
  app.use(express.json());

  app.post("/listen", (req, res) => {
    const { headers, body } = req;

    console.log(headers, body);

    res.sendStatus(200);
  });

  app.get("/", (req, res) => {
    res.status(200).send("coral-webhook-listener");
  });

  app.listen(PORT, HOST, () => {
    console.log(`coral-webhook-listener is listening on ${HOST}:${PORT}...`);
  });
};

void run();
```

- start up Coral
- go to `Admin > Configure > Webhooks` and click `+ ADD WEBHOOK ENDPOINT`
- for `Endpoint URL` use: `http://localhost:7000/listen`
- check on `COMMENT_LEFT_MODERATION_QUEUE`
- click `ADD WEBHOOK ENDPOINT`
- create some comments
- approve some comments
- reject other comments
- check the feed on your listener and see that it has a payload similar to this for each of the comments as you perform approve/reject mod actions:

```
{
  id: '6fb1da28-c468-4829-a1c1-deee0c4e568c',
  createdAt: '2023-10-24T20:52:51.679Z',
  data: {
    queue: 'UNMODERATED',
    commentID: '8f8357e4-1695-422c-af71-ef1bb1923bd6',
    status: 'REJECTED',
    storyID: '1',
    siteID: '4493c905-60e3-4027-8d87-d81a297b74d5'
  },
  type: 'COMMENT_LEFT_MODERATION_QUEUE',
  tenantID: '9f2bd34b-abbc-4602-84e8-91619f5adda4',
  tenantDomain: 'localhost'
}
```

## Where any tests migrated to React Testing Library?

No

## How do we deploy this PR?

No special considerations.
